src/config.h

```diff
@@ -19,99 +19,930 @@
+#endif
+
+    /* ---- Advanced - Video ----*/
+    subsection advanced_video{this, "advanced-video", "Video"};
+
+#ifndef RENDER_CUSTOM
+    enum
+    {
+        RENDER_SOFTWARE = 0,
+        RENDER_ACCELERATED_AUTO,
+        RENDER_ACCELERATED_SDL,
+        RENDER_ACCELERATED_OPENGL,
+        RENDER_ACCELERATED_OPENGL_ES,
+        RENDER_ACCELERATED_OPENGL_LEGACY,
+        RENDER_ACCELERATED_OPENGL_ES_LEGACY,
+        RENDER_END
+    };
+    rendermode_t render_mode{this,
+        {
+            {RENDER_SOFTWARE, "sw", "Software", nullptr},
+            {RENDER_ACCELERATED_AUTO, "hw", "Auto", ""},
+            {RENDER_ACCELERATED_SDL, "sdl", "SDL2", "Basic cross-platform render API"},
+#ifdef THEXTECH_BUILD_GL_DESKTOP_MODERN
+            {RENDER_ACCELERATED_OPENGL, "opengl", "OpenGL 3+", "Desktop API with support for all new visual effects and full accuracy to SMBX64"},
+#endif
+#ifdef THEXTECH_BUILD_GL_ES_MODERN
+            {RENDER_ACCELERATED_OPENGL_ES, "opengles", "OpenGL ES 2+", "Mobile API with support for all new visual effects and high accuracy to SMBX64"},
+#endif
+#ifdef THEXTECH_BUILD_GL_DESKTOP_LEGACY
+            {RENDER_ACCELERATED_OPENGL_LEGACY, "opengl11", "OpenGL 1.1", "Legacy desktop API with full accuracy to SMBX64"},
+#endif
+#ifdef THEXTECH_BUILD_GL_ES_LEGACY
+            {RENDER_ACCELERATED_OPENGL_ES_LEGACY, "opengles11", "OpenGL ES 1.1", "Legacy mobile API with full accuracy to SMBX64"},
+#endif
+            {RENDER_SOFTWARE, "0"},
+            {RENDER_ACCELERATED_SDL, "1"},
+        },
+        defaults(RENDER_ACCELERATED_AUTO), {}, Scope::User,
+        "render", "Render mode", nullptr,
+        config_rendermode_set};
+#endif
+
+#ifndef PGE_MIN_PORT
+    enum ScaleDownTextures
+    {
+        SCALE_DOWN_NONE = 0,
+        SCALE_DOWN_SAFE = 1,
+        SCALE_DOWN_ALL = 2,
+    };
+
+    opt_enum<int> scale_down_textures{this,
+        {
+            {SCALE_DOWN_NONE, "none", "None", "Least loading stutter"},
+            {SCALE_DOWN_SAFE, "safe", "Safe", "Checks if images are in 2x format"},
+            {SCALE_DOWN_ALL, "all", "All", "Less loading stutter than 'Safe'"},
+        },
+        defaults(SCALE_DOWN_SAFE), {}, Scope::User,
+        "scale-down-textures", "Scale down images", "Store images as 1x to save memory"};
+#endif
+
+#ifndef THEXTECH_NO_SDL_BUILD
+    /* ---- Advanced - Audio ----*/
+    subsection advanced_audio{this, "advanced-audio", "Audio"};
+
+    opt<bool> audio_enable{this, defaults(true), {}, Scope::User,
+        "audio-enable", "Enable", nullptr,
+        config_audio_set};
+
+#ifndef _WIN32
+    opt_enum<int> audio_channels{this,
+        {
+            {1, "mono", "Mono", nullptr},
+            {2, "stereo", "Stereo", nullptr},
+        },
+        defaults(g_audioDefaults.channels), {}, Scope::User,
+        "audio-channels", "Channels", nullptr,
+        config_audio_set};
+#else // #ifndef _WIN32
+    static constexpr int audio_channels = 2;
+#endif // #ifndef _WIN32
```

Comment:
Okay, right now the thing is broken: I forgot about some:
- This filed must be writeable, not constexpr (otherwise build fails as there are some places do store defaults on load).
- It's still necessary to allow edit this field, BUT, restrict by manual edit of INI file only, not by GUI.